Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -185,6 +185,39 @@ extension BrowserViewController: WKNavigationDelegate {
     let isPrivateBrowsing = PrivateBrowsingManager.shared.isPrivateBrowsing
     let tab = tabManager[webView]
 
+    let domainForRequestURL = Domain.getOrCreate(
+      forUrl: url,
+      persistent: !isPrivateBrowsing
+    )
+
+    // Debouncing logic
+    // Handle debouncing for main frame only
+    if domainForRequestURL.isShieldExpected(.AdblockAndTp, considerAllShieldsOption: true),
+       navigationAction.targetFrame?.isMainFrame == true,
+       let redirectURL = DebouncingResourceDownloader.shared.redirectURL(for: url) {
```

Comment:
I'd suggest making sure that the redirect URL's scheme is HTTP or HTTPS before debouncing it. The list comes from us and should be okay, but it's not a bad thing to be extra-cautious here.

Reply:
You can do `url.isWebPage(includeDataURIs: false)` for this btw